component/src/main/java/org/wso2/extension/siddhi/io/websocket/sink/WebSocketSink.java

```diff
@@ -146,32 +165,45 @@ protected void init(StreamDefinition streamDefinition, OptionHolder optionHolder
                     throw new SiddhiAppCreationException("The idle timeout defined in '" + streamDefinition +
-                                                                 "' should be greater than 0.");
+                            "' should be greater than 0.");
                 }
             } catch (NumberFormatException e) {
                 throw new SiddhiAppCreationException("The idle timeout defined in '" + streamDefinition +
-                                                             "' should be an Integer.");
+                        "' should be an Integer.");
             }
         }
         connectorListener = new WebSocketClientConnectorListener();
         try {
             String scheme = (new URI(url)).getScheme();
             if (!Objects.equals("ws", scheme) && !Objects.equals("wss", scheme)) {
                 throw new SiddhiAppCreationException("Invalid scheme in " + WebSocketProperties.URL + " = " +
-                                                             url + ". The scheme of the " + WebSocketProperties.URL +
-                                                             " for the websocket server should be either `ws` or "
-                                                             + "`wss`.");
+                        url + ". The scheme of the " + WebSocketProperties.URL +
+                        " for the websocket server should be either `ws` or "
+                        + "`wss`.");
+            }
+            if (Objects.equals("wss", scheme)) {
+                this.sslEnabled = true;
+                this.tlsstruststorePath = optionHolder.validateAndGetStaticValue(
+                        WebSocketProperties.TLS_TRUSTSTORE_PATH, configReader.readConfig
+                                (WebSocketProperties.TLS_TRUSTSTORE_PATH,
+                                        WebSocketProperties
+                                                .DEFAULT_TRUSTSTORE_FILE_PATH));
+                this.tlsstruststorePass = optionHolder.validateAndGetStaticValue(
+                        WebSocketProperties.TLS_TRUSTSTORE_PASS, configReader.readConfig(
+                                WebSocketProperties.TLS_TRUSTSTORE_PASS,
+                                WebSocketProperties.
+                                        DEFAULT_TRUSTSTORE_PASS));
             }
         } catch (URISyntaxException e) {
             throw new SiddhiAppCreationException("There is an syntax error in the '" + url + "' of the websocket "
-                                                         + "server.", e);
+                    + "server.", e);
         }
     }
 
     @Override
     public void connect() throws ConnectionUnavailableException {
         HttpWsConnectorFactory httpConnectorFactory = new DefaultHttpWsConnectorFactory();
-        WsClientConnectorConfig configuration = new WsClientConnectorConfig(url);
+        WebSocketClientConnectorConfig configuration = new WebSocketClientConnectorConfig(url);
```

Comment:
Shall we set `configuration.setAutoRead(true)` since it is not enabled by default and may cause issues.

Reply:
@riyafa, This is a WebSocket client scenario, we will only push messages, we don't read anything from the server.